proto/api/v2/report_configuration_service.proto

```diff
@@ -0,0 +1,118 @@
+    enum ReportType {
+        VULNERABILITY = 0;
+    }
+    string                          id                    = 1;
+    string                          name                  = 2;
+    string                          description           = 3;
+    ReportType                      type                  = 4;
+    oneof                          filter {
+        VulnerabilityReportFilters  vuln_report_filters   = 5;
+    }
+
+    Schedule                        schedule              = 6;
+    ResourceScope                   resource_scope        = 7;
+    repeated NotifierConfiguration  notifiers             = 8;
+}
+
+message VulnerabilityReportFilters {
+    enum Fixability {
+        BOTH        = 0;
+        FIXABLE     = 1;
+        NOT_FIXABLE = 2;
+    }
+
+    enum VulnerabilitySeverity {
+        UNKNOWN_VULNERABILITY_SEVERITY   = 0;
+        LOW_VULNERABILITY_SEVERITY       = 1;
+        MODERATE_VULNERABILITY_SEVERITY  = 2;
+        IMPORTANT_VULNERABILITY_SEVERITY = 3;
+        CRITICAL_VULNERABILITY_SEVERITY  = 4;
+    }
+
+    enum ImageType {
+        DEPLOYED = 0;
+        WATCHED  = 1;
+    }
+
+    Fixability                        fixability              = 1;
+    repeated VulnerabilitySeverity    severities              = 2;
+    repeated ImageType                image_types             = 3;
+    oneof cves_since {
+        bool                          all_vuln                = 4;
+        bool                          last_successful_report  = 5;
+        google.protobuf.Timestamp     start_date              = 6;
+    }
+}
+
+message Schedule {
+    enum IntervalType {
+        UNSET           = 0;
+        DAILY           = 1;
+        WEEKLY          = 2;
+        MONTHLY         = 3;
+    }
+
+    message WeeklyInterval {
+        int32 day = 1;
+    }
+
+    // Sunday = 0, Monday = 1, .... Saturday =  6
+    message DaysOfWeek {
+        repeated int32 days = 1;
+    }
+    // 1 for 1st, 2 for 2nd .... 31 for 31st
+    message DaysOfMonth {
+        repeated int32 days = 1;
+    }
+
+    IntervalType interval_type      = 1;
+    int32        hour               = 2;
+    int32        minute             = 3;
+
+    oneof Interval {
+        WeeklyInterval weekly       = 4;
+        DaysOfWeek days_of_week     = 5;
+        DaysOfMonth days_of_month   = 6;
+    }
+}
+
+message ResourceScope {
+    oneof scope_reference {
+        string collection_id = 1;
+    }
+}
+
+message NotifierConfiguration {
+    oneof notifier_config {
+        EmailNotifierConfiguration email_config = 1;
+    }
+}
+
+message EmailNotifierConfiguration  {
+    string                 notifier_id    = 1;
+    repeated string        mailing_lists  = 2;
+}
+
+
+service ReportConfigurationService {
+    // PostReportConfiguration creates a report configuration
+    rpc PostReportConfiguration (ReportConfiguration) returns (ReportConfiguration) {
+        option (google.api.http) = {
+            post: "/v2/reports/configurations"
```

Comment:
Maybe there is a misunderstanding, but based on our conversation, I thought that we were *not* going to add v2 APIs at this point in time until we have a doc outlining the principles surrounding it

Reply:
@connorgorman The v2 API semantics are still experimental and we are not going to expose these APIs to the users until the reporting feature is shipped. I will be sharing the doc and revisiting the APIs based on comments after implementing and testing the business logic for this feature.